compiler/daml-lf-tools/src/DA/Daml/LF/Simplifier.hs

```diff
@@ -473,9 +473,11 @@ simplifyExpr = fmap fst . cata go'
       e -> (embed (fmap fst e), infoStep world (fmap snd e))
 
 -- | If we have a closed term under a lambda, we want to lift it up to the top level,
--- even though the result of the lambda is also a closed term.
-alwaysLiftUnder :: ExprF t -> Bool
+-- even though the result of the lambda is also a closed term. We avoid breaking up
+-- lambda terms, though.
+alwaysLiftUnder :: ExprF (Expr, Info) -> Bool
 alwaysLiftUnder = \case
+    ETmLamF _ (ETmLam _ _, _) -> False
```

Comment:
Type level lambdas of kind `Nat` are essentially value level lambdas. Any thoughts on how to handle them?

Reply:
Do you mean `ETyLam`? I can add special case to treat `ETyLam (_, KNat) _` as if it were a term lambdas. I wonder if those extra cases are worth the added complexity? Hmm...